### PR TITLE
add UT for pilot/pkg/serviceregistry/kube

### PR DESCRIPTION
**Pull request by stewart-yu**
xref #8568 
before change
```
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.go:197:	Services			      0.0%
.....
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.go:265:	ManagementPorts			      0.0%
```
and after change
```
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.go:197:	Services			      100.0%
...
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.go:265:	ManagementPorts			      71.4%
```

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller_test.go [hunk 1]
```diff
@@ -464,6 +464,44 @@ func TestWorkloadHealthCheckInfoPrometheusPort(t *testing.T) {
...
+			Port:     9090,
+			Protocol: model.ProtocolHTTP,
+		},
+	}
+
+	if len(portList) != len(expected) {
+		t.Errorf("Expecting %d port but got %d\r\n", len(expected), len(portList))
+	}
+
+	for i, exp := range expected {
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller_test.go]: redundant, call` reflect.DeepEqual(expected, portList)`

Comment (stewart-yu) [reply pilot/pkg/serviceregistry/kube/controller_test.go]: seems change like are wrong, need revert

**Pull request merged**